local-dev-environment/search_client.py

```diff
@@ -0,0 +1,117 @@
+                ]
+            }
+        },
+    }
+
+
+# --- 4. Execute results and write to Excel ---
+def local_search_client():
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger("embedding_search_client")
+
+    embedding_generator = EmbeddingGenerator(settings.BEDROCK_EMBEDDING_MODEL_ID)
+    embedding = embedding_generator.generate_embedding(SEARCH_TERM)
+    logger.info(f"Generated embedding for search term: '{SEARCH_TERM}'")
+
+    client = OpenSearch(
+        hosts=[{"host": HOST, "port": PORT}],
+        http_auth=(USER, PASSWORD),
+        use_ssl=False,
+        verify_certs=False,
+        ssl_assert_hostname=False,
+    )
+
+    search_query = create_hybrid_query(SEARCH_TERM, embedding, k=K_QUERIES)
+    logger.info(f"Performing hybrid search for {K_QUERIES} neighbors in '{CHUNK_INDEX_NAME}'...")
+    response = client.search(index=CHUNK_INDEX_NAME, body=search_query)
+
+    hits = response["hits"]["hits"]
+
+    # print(f"hits: {hits}")
+    return hits
+
+
+def write_hits_to_xlsx(hits, score_filter=SCORE_FILTER, search_term=SEARCH_TERM):
+    today_folder = datetime.now().strftime("%Y-%m-%d")
+    output_dir = Path("output/hybrid-test-results") / today_folder
+    output_dir.mkdir(parents=True, exist_ok=True)
+    safe_search_term = str(search_term).replace("/", "_").replace(" ", "_")
+    timestamp_str = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    output_path = output_dir / f"{timestamp_str}_{safe_search_term}_search_results.xlsx"
+    workbook = xlsxwriter.Workbook(str(output_path))
+    worksheet = workbook.add_worksheet()
+    # Write search parameters in the first row
+    worksheet.write(0, 0, "Search term:")
+    worksheet.write(1, 0, search_term)
+    worksheet.write(0, 1, "K_queries:")
+    worksheet.write(1, 1, K_QUERIES)
+    worksheet.write(0, 2, "Score filter:")
+    worksheet.write(1, 2, score_filter)
+    worksheet.write(0, 3, "Keyword boost:")
+    worksheet.write(1, 3, KEYWORD_BOOST)
+    worksheet.write(0, 4, "Semantic boost:")
+    worksheet.write(1, 4, SEMANTIC_BOOST)
+    worksheet.write(0, 5, "No of results")
+    worksheet.write(1, 5, len(hits))
+
+    headers = ["Score", "Case Ref", "Chunk ID", "Page", "Text Snippet"]
+    for col, header in enumerate(headers):
+        worksheet.write(2, col, header)
+    filtered_hits = [hit for hit in hits if hit["_score"] >= score_filter]
+    for row, hit in enumerate(filtered_hits, start=0):
+        score = hit["_score"]
+        source = hit["_source"]
+        worksheet.write(row + 3, 0, score)
+        worksheet.write(row + 3, 1, source.get("case_ref", "N/A"))
+        worksheet.write(row + 3, 2, str(hit.get("_id", "N/A")))
+        worksheet.write(row + 3, 3, source.get("page_number", "N/A"))
+        text_snippet = source.get("chunk_text", "")
+        worksheet.write(row + 3, 4, text_snippet)
+    workbook.close()
+    logging.info(f"Results written to {output_path.resolve()}")
```

Comment:
The log message does not specify the number of results written. Including this information can help with debugging and tracking output size.

Suggested change:
+     logging.info(f"Results written to {output_path.resolve()} ({len(filtered_hits)} results)")